clang/include/clang/Basic/Features.def

```diff
@@ -236,6 +236,7 @@ FEATURE(shadow_call_stack,
 FEATURE(tls, PP.getTargetInfo().isTLSSupported())
 FEATURE(underlying_type, LangOpts.CPlusPlus)
 FEATURE(experimental_library, LangOpts.ExperimentalLibrary)
+FEATURE(datasizeof, LangOpts.CPlusPlus)
```

Comment:
`FEATURE` is for standards features, this should be exposed under `EXTENSION` instead.

Reply:
The problem with `EXTENSION` is that it's influenced by `-pedantic-errors`, which makes it pretty much useless for standard libraries.

Reply:
Yeahhhhh, but we still need to diagnose it as an extension because that's what it is. (We should not add extensions that we do not diagnose under `-pedantic` -- we have been inconsistent in the past but we've gotten better in the past several years.)

Pedantic warnings aren't triggered in system headers, so I think I'm failing to see the concern: https://godbolt.org/z/nn7TdKrcv

Reply:
The problem is that `-pedantic-errors` makes `__has_extension` useless: https://godbolt.org/z/7TfMjf5h6

Reply:
Oh my. That behavior makes sense, but is really not helpful in this case.

You could use compiler version macro checks instead of `__has_extension`. That's pretty gross and likely easy to forget, but would it be a viable workaround?

Reply:
Yeah, that's what we'd have to fall back to - or just live with using the fallback when changing warning flags (!?).

Not really relevant for this patch, but I'm not sure the behaviour actually makes that much sense. There are a few things that make it somewhat weird IMO.
1) `-pedantic-errors` disables `__has_extension`, but `-Werror=pedantic` doesn't. These are clearly distinct flags, but for a user they look like aliases, since they interact with warnings identically (AFAIK).
2) If you use `__has_extension` you almost always want to know whether the compiler accepts some code, and if not you generate an error that you don't support the compiler or use some fallback implementation. (At least that's my interpretation)
3) Changing diagnostic flags shouldn't influence anything other than the behaviour of the diagnostics engine. The code shouldn't compile differently, you should just get different diagnostics.